graylog2-web-interface/src/views/logic/fieldactions/AddToTableActionHandler.js

```diff
@@ -13,15 +10,18 @@ const AddToTableActionHandler: FieldActionHandler = ({ field, contexts: { widget
   return WidgetActions.updateConfig(widget.id, newConfig);
 };
 
-AddToTableActionHandler.isEnabled = ({ contexts: { widget }, field }) => {
-  if (widget instanceof MessagesWidget && widget.config instanceof MessagesWidgetConfig) {
+const isEnabled: FieldActionHandlerCondition = ({ contexts: { widget }, field }) => {
+  if (widget.type === 'messages' && widget.config) {
```

Comment:
We could improve this in two ways:

  1.) Extract the check into a `isMessageWidget(widget)` function in a separate module and reuse it.
  2.) Use the `MessagesWidget.type` constant

The check generally is okay. The only concern that I have is that we _might_ have subclasses in the future with a different type forcing us to take care of refining the check. That is why I am suggesting to extract the function, to make it more visible.

Reply:
Good point, I implemented this check inside `MessagesWidget` is it fine or do you have a better place in mind?